### PR TITLE
Numpy

### PR DESCRIPTION
**Pull request by pelson**
Iterates on #517.
I'm probably not going to be able to iterate on this in all honesty, but I wanted to push it to see how it runs on CI.

### CODE REVIEW

**Comment by jakirkham** [comment]
So, as we don't have OpenBLAS on Windows yet. Is there a way to skip it?

File: recipes/numpy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  name: numpy
+  version: {{ version }}
+
+source:
+  fn: numpy-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/n/numpy/numpy-{{ version }}.tar.gz
+  md5: bc56fb9fc2895aa4961802ffbdb31d0b
+
+build:
+  number: 0
```
Comment (jakirkham) [inline recipes/numpy/meta.yaml]: Probably should start with something high.

File: recipes/numpy/build.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+[openblas]
+libraries = openblas
+library_dirs = $PREFIX/lib
+include_dirs = $PREFIX/include
+siteconfig
+
+$PYTHON setup.py config
+$PYTHON setup.py build -j $CPU_COUNT
+$PYTHON setup.py install --old-and-unmanageable
```
Comment (jakirkham) [inline recipes/numpy/build.sh]: `--single-version-externally-managed --record=record.txt`

File: recipes/numpy/bld.bat [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+echo atlas_libs = openblas
+echo libraries = openblas
+echo.
+echo [openblas]
+echo libraries = openblas
+echo library_dirs = %LIBRARY_LIB%
+echo include_dirs = %LIBRARY_INC%
+) > site.cfg
+
+python setup.py build -j %CPU_COUNT% install --old-and-unmanageable
```
Comment (jakirkham) [inline recipes/numpy/bld.bat]: `--single-version-externally-managed --record=record.txt`

File: recipes/numpy/build.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+libraries = openblas
+
+[openblas]
+libraries = openblas
+library_dirs = $PREFIX/lib
+include_dirs = $PREFIX/include
+siteconfig
+
+$PYTHON setup.py config
+$PYTHON setup.py build -j $CPU_COUNT
```
Comment (jakirkham) [inline recipes/numpy/build.sh]: `CPU_COUNT` is a bit dubious with CI as they often misrepresent this figure. For instance, they might tell you how many cores there are on the node you are, but you may not be allowed to use that many. Normally, I just drop it entirely. In the case of `numpy`, it builds so fast I have never felt a need for this.

File: recipes/numpy/bld.bat (hunk 3 above)
Comment (jakirkham) [inline recipes/numpy/bld.bat]: Guessing we will need a `config` and an `install` step.

**Comment by pelson** [comment]
Not sure how well this is doing (Linux):

```
Running from numpy source directory.
blas_opt_info:
blas_mkl_info:
/opt/conda/conda-bld/work/numpy-1.11.0/numpy/distutils/system_info.py:633: UserWarning: Specified path $PREFIX/lib is invalid.
  warnings.warn('Specified path %s is invalid.' % d)
/opt/conda/conda-bld/work/numpy-1.11.0/numpy/distutils/system_info.py:633: UserWarning: Specified path $PREFIX/include is invalid.
  warnings.warn('Specified path %s is invalid.' % d)
  libraries mkl,vml,guide not found in []
  NOT AVAILABLE

openblas_info:
  libraries openblas not found in []
  NOT AVAILABLE

atlas_3_10_blas_threads_info:
Setting PTATLAS=ATLAS
  libraries openblas not found in []
  NOT AVAILABLE

atlas_3_10_blas_info:
  libraries openblas not found in []
  NOT AVAILABLE

atlas_blas_threads_info:
Setting PTATLAS=ATLAS
  libraries openblas not found in []
  NOT AVAILABLE

atlas_blas_info:
  libraries openblas not found in []
  NOT AVAILABLE

/opt/conda/conda-bld/work/numpy-1.11.0/numpy/distutils/system_info.py:1640: UserWarning: 
    Atlas (http://math-atlas.sourceforge.net/) libraries not found.
    Directories to search for the libraries can be specified in the
    numpy/distutils/site.cfg file (section [atlas]) or by setting
    the ATLAS environment variable.
  warnings.warn(AtlasNotFoundError.__doc__)
blas_info:
  libraries blas not found in []
  NOT AVAILABLE

/opt/conda/conda-bld/work/numpy-1.11.0/numpy/distutils/system_info.py:1649: UserWarning: 
    Blas (http://www.netlib.org/blas/) libraries not found.
    Directories to search for the libraries can be specified in the
    numpy/distutils/site.cfg file (section [blas]) or by setting
    the BLAS environment variable.
  warnings.warn(BlasNotFoundError.__doc__)
blas_src_info:
  NOT AVAILABLE

/opt/conda/conda-bld/work/numpy-1.11.0/numpy/distutils/system_info.py:1652: UserWarning: 
    Blas (http://www.netlib.org/blas/) sources not found.
    Directories to search for the sources can be specified in the
    numpy/distutils/site.cfg file (section [blas_src]) or by setting
    the BLAS_SRC environment variable.
  warnings.warn(BlasSrcNotFoundError.__doc__)
  NOT AVAILABLE

/bin/sh: svnversion: command not found
/bin/sh: svnversion: command not found
F2PY Version 2
lapack_opt_info:
openblas_lapack_info:
  libraries openblas not found in []
  NOT AVAILABLE

lapack_mkl_info:
mkl_info:
  libraries mkl,vml,guide not found in []
  NOT AVAILABLE

  NOT AVAILABLE

atlas_3_10_threads_info:
Setting PTATLAS=ATLAS
<class 'numpy.distutils.system_info.atlas_3_10_threads_info'>
  NOT AVAILABLE

atlas_3_10_info:
<class 'numpy.distutils.system_info.atlas_3_10_info'>
  NOT AVAILABLE

atlas_threads_info:
Setting PTATLAS=ATLAS
<class 'numpy.distutils.system_info.atlas_threads_info'>
  NOT AVAILABLE

atlas_info:
<class 'numpy.distutils.system_info.atlas_info'>
  NOT AVAILABLE

/opt/conda/conda-bld/work/numpy-1.11.0/numpy/distutils/system_info.py:1542: UserWarning: 
    Atlas (http://math-atlas.sourceforge.net/) libraries not found.
    Directories to search for the libraries can be specified in the
    numpy/distutils/site.cfg file (section [atlas]) or by setting
    the ATLAS environment variable.
  warnings.warn(AtlasNotFoundError.__doc__)
lapack_info:
  libraries lapack not found in []
  NOT AVAILABLE

/opt/conda/conda-bld/work/numpy-1.11.0/numpy/distutils/system_info.py:1553: UserWarning: 
    Lapack (http://www.netlib.org/lapack/) libraries not found.
    Directories to search for the libraries can be specified in the
    numpy/distutils/site.cfg file (section [lapack]) or by setting
    the LAPACK environment variable.
  warnings.warn(LapackNotFoundError.__doc__)
lapack_src_info:
  NOT AVAILABLE

/opt/conda/conda-bld/work/numpy-1.11.0/numpy/distutils/system_info.py:1556: UserWarning: 
    Lapack (http://www.netlib.org/lapack/) sources not found.
    Directories to search for the sources can be specified in the
    numpy/distutils/site.cfg file (section [lapack_src]) or by setting
    the LAPACK_SRC environment variable.
  warnings.warn(LapackSrcNotFoundError.__doc__)
  NOT AVAILABLE
```


@jakirkham - do you want to take the mantle forwards on this recipe?

**Comment by pelson** [comment]
OSX: 

```

Running from numpy source directory.

setup.py:327: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates

  warnings.warn("Unrecognized setuptools command, proceeding with "

blas_opt_info:

blas_mkl_info:

/Users/travis/miniconda/conda-bld/work/numpy-1.11.0/numpy/distutils/system_info.py:633: UserWarning: Specified path $PREFIX/lib is invalid.

  warnings.warn('Specified path %s is invalid.' % d)

/Users/travis/miniconda/conda-bld/work/numpy-1.11.0/numpy/distutils/system_info.py:633: UserWarning: Specified path $PREFIX/include is invalid.

  warnings.warn('Specified path %s is invalid.' % d)

  libraries mkl,vml,guide not found in []

  NOT AVAILABLE



openblas_info:

  libraries openblas not found in []

  NOT AVAILABLE



atlas_3_10_blas_threads_info:

Setting PTATLAS=ATLAS

  libraries openblas not found in []

  NOT AVAILABLE



atlas_3_10_blas_info:

  libraries openblas not found in []

  NOT AVAILABLE



atlas_blas_threads_info:

Setting PTATLAS=ATLAS

  libraries openblas not found in []

  NOT AVAILABLE



atlas_blas_info:

  libraries openblas not found in []

  NOT AVAILABLE



  FOUND:

    extra_link_args = ['-Wl,-framework', '-Wl,Accelerate']

    define_macros = [('NO_ATLAS_INFO', 3), ('HAVE_CBLAS', None)]

    extra_compile_args = ['-msse3', '-I/System/Library/Frameworks/vecLib.framework/Headers']



F2PY Version 2

lapack_opt_info:

openblas_lapack_info:

  libraries openblas not found in []

  NOT AVAILABLE



lapack_mkl_info:

mkl_info:

  libraries mkl,vml,guide not found in []

  NOT AVAILABLE



  NOT AVAILABLE



atlas_3_10_threads_info:

Setting PTATLAS=ATLAS

<class 'numpy.distutils.system_info.atlas_3_10_threads_info'>

  NOT AVAILABLE



atlas_3_10_info:

<class 'numpy.distutils.system_info.atlas_3_10_info'>

  NOT AVAILABLE



atlas_threads_info:

Setting PTATLAS=ATLAS

<class 'numpy.distutils.system_info.atlas_threads_info'>

  NOT AVAILABLE



atlas_info:

<class 'numpy.distutils.system_info.atlas_info'>

  NOT AVAILABLE



  FOUND:

    extra_link_args = ['-Wl,-framework', '-Wl,Accelerate']

    define_macros = [('NO_ATLAS_INFO', 3), ('HAVE_CBLAS', None)]

    extra_compile_args = ['-msse3']
```

That isn't what we want (accelerate).

**Comment by pelson** [comment]
Think this is now resolved (it is on my machine anyway). The `$PREFIX` was being escaped and not expanded, so numpy wasn't able to find an openblas in the `\$PREFIX` directory 😄

**Comment by jakirkham** [comment]
@charris @njsmith @matthew-brett @rgommers @stefanv @mwiebe @jaimefrio @juliantaylor @seberg, just letting you know that we have added `numpy` to [conda-forge]( http://conda-forge.github.io/ ). This will allow users to easily install this with [`conda`]( http://conda.pydata.org/docs/ ) ( [`conda install -c conda-forge numpy`]( http://anaconda.org/conda-forge/numpy ) ). The maintenance burden here is light as all releases are built and deployed automatically with CI. Changes to how it is being built can be controlled by changing the recipe in [this feedstock (repo)]( https://github.com/conda-forge/numpy-feedstock ). If you have any interest in being added as a maintainer (to make new releases for instance), we would be more than happy to add any of you are interested to the maintainers list.

**Pull request merged**